pages/administratorWorkspaces/administratorWorkspaces.sql

```diff
@@ -0,0 +1,28 @@
+-- BLOCK select_workspaces
+SELECT
+  w.id,
+  w.state,
+  format_interval (
+    CASE
+      WHEN w.state = 'launching' THEN (NOW() - w.launched_at)
+      ELSE (NOW() - w.running_at)
+    END
+  ) AS time_in_state,
+  wh.id AS workspace_host_id,
+  wh.instance_id AS workspace_host_instance_id,
+  wh.hostname AS workspace_host_hostname,
+  wh.state AS workspace_host_state,
+  format_interval (
+    CASE
+      WHEN wh.state = 'launching' THEN (NOW() - wh.launched_at)
+      WHEN wh.state = 'ready' THEN (NOW() - wh.ready_at)
+      WHEN wh.state = 'unhealthy' THEN (NOW() - wh.unhealthy_at)
+      WHEN wh.state = 'terminated' THEN (NOW() - wh.terminated_at)
+      ELSE make_interval()
+    END
+  ) AS workspace_host_time_in_state
```

Comment:
I'd like to start moving towards:
1. Exporting whole "objects" from the DB, so this query would just be `to_json(w) AS workspace, to_json(wh) AS workspace_host`, and having the equivalent of the PT `db-types.ts` file with types already defined for each table.
2. Doing date/interval formatting in JS, by packaging up the PT formatting functions.

No need to do this now for this PR (unless you want to) but I'm just putting this on the record for discussion.

Reply:
I know we do that in PT, but I'm wary about that because of the overfetching problem (I forget if we've discussed this before in PT). There's a mostly small but very real CPU/network/memory cost to indiscriminately fetching all columns; we've seen that in an extreme case recently with `issues`/`submissions`.

Agreed on doing formatting in JS; the only reason I didn't was because we don't have all our nice helper functions from PT yet!